docs/csharp/codedoc.md

```diff
@@ -2,26 +2,1348 @@
->
+# Documenting your code
+
+Documenting C# code is recommended practice as it helps make code easy to understand and maintain.
+Comments are a great way to add documentation to your C# code, they can help explain parts of your code and the great thing is,
+they do not add any overhead to your workflow because they usually are completely ignored by the compiler.
+
+## Single Line Comments
+
+Single line comments are a great way to explain C# code. They can be on a separate line or inline with your code.
+The comment body is prefixed by double forward slashes (`//`).
+
+```csharp
+// This comment describes variable i
+// Multiple lines could be a hassle
+int i = 0;
+int j = 1; // This comment describes variable j
+```
+
+## Multiline Comments
+
+Multiline comments function just like single line comments except that they make it easier to have comments that span multiple lines.
+The comment body starts with (`/*`) and ends with (`*/`).
+
+```csharp
+/*
+This describes variables i and j
+Multiline is less of a hassle
+*/
+int i = 0, j = 1;
+```
+
+Single line and Multiline comments are very useful for explaining code and getting new contributors up to speed with the codebase. They however 
+do not account for situations where the .NET assembly is distributed as a binary to third party developers with no available source code.
+This is where XML documentation comments come in handy.
+
+
+## XML Documentation Comments
+
+XML documentation comments are a special kind of comment, added above the definition of any user defined type or member. 
+They are special because they can be processed by the compiler to generate an XML documentation file at compile time.
+The compiler generated XML file can be distributed alongside your .NET assembly so that Visual Studio and other IDEs can show quick information about types or members when performing intellisense.
+Additionally the XML file can be run through tools like [DocFX](https://dotnet.github.io/docfx/) and [Sandcastle](https://github.com/EWSoftware/SHFB) to generate full on API reference websites.
+
+XML documentation comments like all other comments are ignored by the compiler, to enable generation of the XML file add `"xmlDoc":true` under `buildOptions` in your `project.json` when using .NET Core or use the `/doc` compiler option for the .NET framework.
+Go [here](https://msdn.microsoft.com/en-us/library/3260k4x7.aspx) to learn how to enable XML documentation generation in Visual Studio.
+
+XML documentation comments are characterized by triple forward slashes (`///`) and an XML formatted comment body.
+
+```csharp
+/// <summary>
+/// This class does something.
+/// </summary>
+public class SomeClass
+{
+
+}
+```
+
+## Walkthrough
+
+In this section I'm going to walk you through documenting a very basic math library to make it easy for new developers to understand/contribute and for third party developers to use.
```

Comment:
We use 2nd person (refer to the reader as 'you') throughout the site, instead of first person (using 'I'). This is the first example. I'll try to catch all of them throughout the article.